python3Packages.aioesphomeapi: 41.13.0 -> 42.0.0

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/nix-community/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.

meta.description for python3Packages.aioesphomeapi is: Python Client for ESPHome native API

meta.homepage for python3Packages.aioesphomeapi is: https://github.com/esphome/aioesphomeapi

meta.changelog for python3Packages.aioesphomeapi is: https://github.com/esphome/aioesphomeapi/releases/tag/v42.0.0


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 42.0.0 with grep in /nix/store/dhfjj6r7g9xm4n88scxr8kg1m8xgar6v-python3.13-aioesphomeapi-42.0.0
- found 42.0.0 in filename of file in /nix/store/dhfjj6r7g9xm4n88scxr8kg1m8xgar6v-python3.13-aioesphomeapi-42.0.0

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
55 total rebuild path(s)

55 package rebuild(s)

First fifty rebuilds by attrpath

esphome
home-assistant-component-tests.acaia
home-assistant-component-tests.airthings_ble
home-assistant-component-tests.aranet
home-assistant-component-tests.bluemaestro
home-assistant-component-tests.bluetooth_adapters
home-assistant-component-tests.bluetooth_le_tracker
home-assistant-component-tests.bthome
home-assistant-component-tests.dormakaba_dkey
home-assistant-component-tests.eq3btsmart
home-assistant-component-tests.esphome
home-assistant-component-tests.eufylife_ble
home-assistant-component-tests.fjaraskupan
home-assistant-component-tests.gardena_bluetooth
home-assistant-component-tests.govee_ble
home-assistant-component-tests.homekit_controller
home-assistant-component-tests.husqvarna_automower_ble
home-assistant-component-tests.ibeacon
home-assistant-component-tests.idasen_desk
home-assistant-component-tests.improv_ble
home-assistant-component-tests.inkbird
home-assistant-component-tests.iron_os
home-assistant-component-tests.kegtron
home-assistant-component-tests.keymitt_ble
home-assistant-component-tests.kulersky
home-assistant-component-tests.lamarzocco
home-assistant-component-tests.ld2410_ble
home-assistant-component-tests.leaone
home-assistant-component-tests.led_ble
home-assistant-component-tests.medcom_ble
home-assistant-component-tests.melnor
home-assistant-component-tests.moat
home-assistant-component-tests.mopeka
home-assistant-component-tests.motionblinds_ble
home-assistant-component-tests.oralb
home-assistant-component-tests.private_ble_device
home-assistant-component-tests.probe_plus
home-assistant-component-tests.qingping
home-assistant-component-tests.rapt_ble
home-assistant-component-tests.ruuvitag_ble
home-assistant-component-tests.sensirion_ble
home-assistant-component-tests.sensorpro
home-assistant-component-tests.sensorpush
home-assistant-component-tests.snooz
home-assistant-component-tests.switchbot
home-assistant-component-tests.thermobeacon
home-assistant-component-tests.thermopro
home-assistant-component-tests.tilt_ble
home-assistant-component-tests.xiaomi_ble
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/dhfjj6r7g9xm4n88scxr8kg1m8xgar6v-python3.13-aioesphomeapi-42.0.0 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A python3Packages.aioesphomeapi https://github.com/r-ryantm/nixpkgs/archive/a3848316a908d5a22d8f9fcfe2f17b4151186131.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/a3848316a908d5a22d8f9fcfe2f17b4151186131#python3Packages.aioesphomeapi
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/dhfjj6r7g9xm4n88scxr8kg1m8xgar6v-python3.13-aioesphomeapi-42.0.0
ls -la /nix/store/dhfjj6r7g9xm4n88scxr8kg1m8xgar6v-python3.13-aioesphomeapi-42.0.0/bin
```

---

</details>
<br/>



### Pre-merge build results

We have automatically built all packages that will get rebuilt due to
this change.

This gives evidence on whether the upgrade will break dependent packages.
Note sometimes packages show up as _failed to build_ independent of the
change, simply because they are already broken on the target branch.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review --extra-nixpkgs-config '{ allowBroken = false; }'`
Commit: `a3848316a908d5a22d8f9fcfe2f17b4151186131`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 59 packages built:</summary>
  <ul>
    <li>esphome</li>
    <li>esphome.dist</li>
    <li>home-assistant-component-tests.acaia</li>
    <li>home-assistant-component-tests.airthings_ble</li>
    <li>home-assistant-component-tests.aranet</li>
    <li>home-assistant-component-tests.bluemaestro</li>
    <li>home-assistant-component-tests.bluetooth_adapters</li>
    <li>home-assistant-component-tests.bluetooth_le_tracker</li>
    <li>home-assistant-component-tests.bthome</li>
    <li>home-assistant-component-tests.dormakaba_dkey</li>
    <li>home-assistant-component-tests.eq3btsmart</li>
    <li>home-assistant-component-tests.esphome</li>
    <li>home-assistant-component-tests.eufylife_ble</li>
    <li>home-assistant-component-tests.fjaraskupan</li>
    <li>home-assistant-component-tests.gardena_bluetooth</li>
    <li>home-assistant-component-tests.govee_ble</li>
    <li>home-assistant-component-tests.homekit_controller</li>
    <li>home-assistant-component-tests.husqvarna_automower_ble</li>
    <li>home-assistant-component-tests.ibeacon</li>
    <li>home-assistant-component-tests.idasen_desk</li>
    <li>home-assistant-component-tests.improv_ble</li>
    <li>home-assistant-component-tests.inkbird</li>
    <li>home-assistant-component-tests.iron_os</li>
    <li>home-assistant-component-tests.kegtron</li>
    <li>home-assistant-component-tests.keymitt_ble</li>
    <li>home-assistant-component-tests.kulersky</li>
    <li>home-assistant-component-tests.lamarzocco</li>
    <li>home-assistant-component-tests.ld2410_ble</li>
    <li>home-assistant-component-tests.leaone</li>
    <li>home-assistant-component-tests.led_ble</li>
    <li>home-assistant-component-tests.medcom_ble</li>
    <li>home-assistant-component-tests.melnor</li>
    <li>home-assistant-component-tests.moat</li>
    <li>home-assistant-component-tests.mopeka</li>
    <li>home-assistant-component-tests.motionblinds_ble</li>
    <li>home-assistant-component-tests.oralb</li>
    <li>home-assistant-component-tests.private_ble_device</li>
    <li>home-assistant-component-tests.probe_plus</li>
    <li>home-assistant-component-tests.qingping</li>
    <li>home-assistant-component-tests.rapt_ble</li>
    <li>home-assistant-component-tests.ruuvitag_ble</li>
    <li>home-assistant-component-tests.sensirion_ble</li>
    <li>home-assistant-component-tests.sensorpro</li>
    <li>home-assistant-component-tests.sensorpush</li>
    <li>home-assistant-component-tests.snooz</li>
    <li>home-assistant-component-tests.switchbot</li>
    <li>home-assistant-component-tests.thermobeacon</li>
    <li>home-assistant-component-tests.thermopro</li>
    <li>home-assistant-component-tests.tilt_ble</li>
    <li>home-assistant-component-tests.xiaomi_ble</li>
    <li>home-assistant-component-tests.yalexs_ble</li>
    <li>python312Packages.aioesphomeapi</li>
    <li>python312Packages.aioesphomeapi.dist</li>
    <li>python312Packages.bleak-esphome</li>
    <li>python312Packages.bleak-esphome.dist</li>
    <li>python313Packages.aioesphomeapi</li>
    <li>python313Packages.aioesphomeapi.dist</li>
    <li>python313Packages.bleak-esphome</li>
    <li>python313Packages.bleak-esphome.dist</li>
  </ul>
</details>

---

###### Maintainer pings

cc @fabaff @mweinelt for [testing](https://github.com/nix-community/nixpkgs-update/blob/main/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

r-ryantm committed:
python3Packages.aioesphomeapi: 41.13.0 -> 42.0.0

sarahec reviewed (changes requested):
Need to disable `tests/test_host_resolver.py` on Darwin due to lack of network namespacing. (Python 3.12 and 3.13 builds collide and one of them fails.)

mweinelt committed:
python3Packages.aioesphomeapi: disable tests on darwin

Due to the lack of network sandboxing on darwin parallel builds of this
package, e.g. during nixpkgs-review, leads to conflicting socket binds,
which makes one of the packages fail.

This issues stretches over multiple files, so that it is tedious to
disasble these tests selectivly.

mweinelt commented:
Found even more, opted to disable the tests on darwin instead.
